src/components/SettingsModal.tsx

```diff
@@ -68,10 +106,19 @@ export const SettingsModal = ({ handleCloseModal, isModalOpen }: ISettingsModal)
 
   const onSave = () => {
     setSettings(localSettings);
+    const flowSettings = Object.entries(settings).filter(
+      ([key]) => key !== 'name' && key !== 'description',
+    );
+    if (localSettings.name != null) {
+      metadata.name = localSettings.name;
+    }
+    if (localSettings.description != null) {
+      metadata.description = localSettings.description;
+    }
     const updatedFlowWrapper = {
       flows: flows.map((flow) => ({
         ...flow,
-        metadata: { ...flow.metadata, ...settings },
+        metadata: { ...flow.metadata, ...flowSettings },
```

Comment:
Deestructuring an array into an object brings the arrays keys as object keys, check the following examples: 

![image](https://github.com/KaotoIO/kaoto-ui/assets/16512618/0cb427ea-c5da-484a-b5ee-c0f01a9b188d)

![image](https://github.com/KaotoIO/kaoto-ui/assets/16512618/3b15adc3-f61e-4d04-9fce-2b4af9f92e60)

I think that we don't need this line at all, because in the past we were setting the whole `settings` object into the metadata, which wasn't correct.
```suggestion
```

Reply:
OK, so we won't set `flow[n].metadata` at all from settings modal, correct? that's much better.

Reply:
oh but don't we bring in what is already there? just put the `flows[n].metadata` as is

Reply:
nevermind... the line `...flow` right above does that

Reply:
done :+1: